src/components/Navbar/TransactionsList.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  CaretDownOutlined,
+  CheckCircleOutlined,
+  CloseCircleOutlined,
+  ExclamationCircleOutlined,
+  ThunderboltOutlined,
+} from '@ant-design/icons'
+import { Trans } from '@lingui/macro'
+import { ThemeContext } from 'contexts/themeContext'
+import { NetworkName } from 'models/network-name'
+import { TxStatus } from 'models/transaction'
+import { CSSProperties, useContext, useEffect, useMemo, useState } from 'react'
+import { formatHistoricalDate } from 'utils/formatDate'
+import { readNetwork } from '../../constants/networks'
+import {
+  timestampForTxLog,
+  TxHistoryContext,
+} from '../../contexts/txHistoryContext'
+import Loading from '../Loading'
+
+export default function TransactionsList({
+  style,
+  listStyle,
+}: {
+  style?: CSSProperties
+  listStyle?: CSSProperties
+}) {
+  const { transactions, removeTransaction } = useContext(TxHistoryContext)
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const [isExpanded, setIsExpanded] = useState<boolean>()
+
+  const TxStatusElem = (status: TxStatus) => {
+    switch (status) {
+      case TxStatus.pending:
+        return <Loading size="small" />
+      case TxStatus.success:
+        return (
+          <div style={{ color: colors.icon.success }}>
+            <CheckCircleOutlined />
+          </div>
+        )
+      case TxStatus.failed:
+        return (
+          <div style={{ color: colors.icon.failure }}>
+            <ExclamationCircleOutlined />
+          </div>
+        )
+    }
+  }
+
+  const hasPendingTxs = useMemo(
+    () => transactions?.some(tx => tx.status === TxStatus.pending),
+    [transactions],
+  )
+
+  useEffect(() => {
+    // Auto expand if pending tx is added
+    if (hasPendingTxs) setIsExpanded(true)
+  }, [hasPendingTxs])
+
+  return (
+    <div style={{ ...style }}>
+      {!!transactions?.length && (
```

Comment:
why the `!!` usage?

Reply:
`transactions?.length` doesn't automatically resolve to boolean, i'm guessing due to the undefined but seems like a ts bug. need to do `transactions && transactions.length`, so this is just shorthand

Reply:
edit—eh nvm i was wrong. i swear i was seeing that before, but looking now it seems like i can remove the `!!`

Reply:
nvm again, just actually remembered why we need this. in tsx && statements, if the first expression is a number `0`, react just renders the 0 instead of evaluating it as boolean. kinda unexpected, but makes sense i guess. keeping the `!!`

<img width="147" alt="image" src="https://user-images.githubusercontent.com/79433522/189168570-caf2ae63-bf62-4319-82c1-9a43f3b50507.png">